### PR TITLE
test: E2E framework (version 2)

### PR DESCRIPTION
**Pull request by D-Reckoning**
## Description
Adds version 2 of the Selenium based E2E testing framework. Code review and feedback welcomed.
Scope: 
- Currently contains framework only.
- CI integration not yet implemented (pending new CI pipeline)

### CODE REVIEW

File: src/Intranet.Selenium.Framework/Element.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using OpenQA.Selenium;
+
+namespace Intranet.Selenium.Framework
+{
+    public class Element
+    {
+        public List<IWebElement> Elements { get; set; }
```
Comment (OskarKlintrot) [inline src/Intranet.Selenium.Framework/Element.cs]: Should use interface, ie `IEnumarable` or `IList` (if you actually needs specifically a List).

File: src/Intranet.Selenium.Framework/Element.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        public Element (IWebElement element, string elementName)
+        {
+            Elements = new List<IWebElement>
+            {
+                element
+            };
+            Name = elementName;
+        }
+
+        public Element (List<IWebElement> elements, string elementName)
```
Comment (OskarKlintrot) [inline src/Intranet.Selenium.Framework/Element.cs]: Same as above, should accept an interface.

File: src/Intranet.Selenium.Framework/ResultTracker.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Intranet.Selenium.Framework
+{
+    public enum AssertLevel
```
Comment (OskarKlintrot) [inline src/Intranet.Selenium.Framework/ResultTracker.cs]: Should be in it's own file: One file, one class.

File: src/Intranet.Selenium.Framework/SeleniumDriver.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Firefox;
+using OpenQA.Selenium.IE;
+using OpenQA.Selenium.PhantomJS;
+using Microsoft.DotNet.PlatformAbstractions;
+using System;
+
+namespace Intranet.Selenium.Framework
+{
+    public enum Browser
```
Comment (OskarKlintrot) [inline src/Intranet.Selenium.Framework/SeleniumDriver.cs]: Same as above.

File: src/Intranet.Selenium.Framework/SeleniumDriver.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            Interact = new SeleniumInteract(_driver, Log);
+            
+        }
+
+        public IWebDriver DriverComponent()
+        {
+            return _driver;
+        }
+
+        public void Kill()
```
Comment (OskarKlintrot) [inline src/Intranet.Selenium.Framework/SeleniumDriver.cs]: https://open.spotify.com/track/2Q4mRo3ZqWHUCKn4vsnyKJ 😄

File: src/Intranet.Selenium.Framework/SeleniumLogger.cs [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.DotNet.PlatformAbstractions;
+using System.IO;
+using OpenQA.Selenium;
+
+namespace Intranet.Selenium.Framework
+{
+    public enum Level
```
Comment (OskarKlintrot) [inline src/Intranet.Selenium.Framework/SeleniumLogger.cs]: You know the drill 😃

File: src/Intranet.Selenium.Tests/FrameworkDiagnostics.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+using OpenQA.Selenium;
+using Intranet.Selenium.Framework;
+using System.Linq;
+
+namespace Intranet.Selenium.Tests
+{
+#pragma warning disable S3881 //full implementation of IDisposable not necessary; only used as a workaround (since xUnit does not support [TearDown] or similar)
```
Comment (OskarKlintrot) [inline src/Intranet.Selenium.Tests/FrameworkDiagnostics.cs]: Can't you use a test fixture instead? I think that is prefererad by xUnit.

File: src/Intranet.Selenium.Framework/Enums/AssertLevel.cs [hunk 8]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Intranet.Selenium.Framework
```
Comment (OskarKlintrot) [inline src/Intranet.Selenium.Framework/Enums/AssertLevel.cs]: You forgot the namespace ;)

File: src/Intranet.Selenium.Tests/FrameworkDiagnostics.cs (hunk 7 above)
Comment (D-Reckoning) [inline src/Intranet.Selenium.Tests/FrameworkDiagnostics.cs]: This is actually the official recommended workaround: https://xunit.github.io/docs/comparisons.html

**Pull request merged**